lib/clarivate/links_client.rb

```diff
@@ -0,0 +1,65 @@
+module Clarivate
+  # Links AMR (Article Match Retrieval) Service
+  # @see http://ipscience-help.thomsonreuters.com/LAMRService/WebServicesOverviewGroup/overview.html Service documentation
+  class LinksClient
+    attr_reader :username, :password, :host
+
+    def initialize(username: nil, password: nil, host: 'https://ws.isiknowledge.com')
+      @host = host
+      @username = username
+      @password = password
+      (@username, @password) = Base64.decode64(Settings.WOS.AUTH_CODE).split(':', 2) unless username
+    end
+
+    # limited to 50 ids
+    # @param [Array<String>] ids
+    # @return [Hash<String => Hash>]
+    def links(ids)
+      raise ArgumentError, "1-50 ids required" unless ids.present? && ids.count <= 50
+      response = connection.send(:post) do |req|
+        req.path = '/cps/xrpc'
+        req.body = request_body(ids.uniq)
+      end
+      ng = Nokogiri::XML(response.body) { |config| config.strict.noblanks }.remove_namespaces!
+      pairs = ng.xpath('response/fn/map/map').map do |node|
+        [node.attr('name').split('_').last, vals_to_hash(node.xpath('//val'))]
+      end
+      pairs.to_h
+    end
+
```

Comment:
What's the logic in separating the `@host` and the `req.path` - can we put the path into the connection object on init (in the `connection` method)?  If the `path` is a constant, a private `PATH = x.freeze` is the place for it; if it might be a config option, `Settings.WOS.LINKS_PATH` could be the place for it.  Consider the config `Settings.WOS.LINKS_HOST` to store the default `host` location (most of the service API configs do this, IIRC).

We're assuming that the LINKS has the same credentials as the SOAP-API here and, AFAIK, that's correct.

Reply:
Since it is actually working, that is demonstrably correct.  Don't care about path being a constant, since it can be overridden if desired (e.g. using a proxy).  Also, as far as this level of the code is concerned, it doesn't need to live in config for any reason.  The specs literally declare what the endpoint is, so it is fine to implement it that way.  If we need to override it, *then* we can worry about where to store the override value.  Pretty sure that will never happen though.